commit f7a59c71d60db3085e927e48665805a3d8c5777c
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Fri Jul 3 16:17:51 2009 +0000

    Check for libdrm >= 2.4 (needed for i965_drv_video.so).

commit 2dce03bcaf13fb5e5125a469d6d2bb473495d082
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jul 2 09:17:45 2009 +0000

    Allow conditional build of video drivers (dummy, i965).

diff --git a/Makefile.am b/Makefile.am
index 189cba8..4f5c787 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -21,7 +21,14 @@
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = src dummy_drv_video i965_drv_video test
+
+SUBDIRS = src test
+if BUILD_DUMMY_DRIVER
+SUBDIRS += dummy_drv_video
+endif
+if BUILD_I965_DRIVER
+SUBDIRS += i965_drv_video
+endif
 
 pcfiles = \
 	libva.pc \
diff --git a/configure.ac b/configure.ac
index 9e6d895..b6e82d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,6 +66,17 @@ LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION -release $LIBVA_VERSION.$LIBVA
 AC_SUBST(LIBVA_LT_VERSION)
 AC_SUBST(LIBVA_LT_LDFLAGS)
 
+AC_ARG_ENABLE(dummy-driver,
+              [AC_HELP_STRING([--enable-dummy-driver],
+                              [build dummy video driver])],
+              [], [enable_dummy_driver=yes])
+AM_CONDITIONAL(BUILD_DUMMY_DRIVER, test x$enable_dummy_driver = xyes)
+
+AC_ARG_ENABLE(i965-driver,
+              [AC_HELP_STRING([--enable-i965-driver],
+                              [build i965 video driver])],
+              [], [enable_i965_driver=yes])
+
 AC_ARG_WITH(drivers-path,
 	    [AC_HELP_STRING([--with-drivers-path=[[path]]], [drivers path])],,
 	    [with_drivers_path="$libdir/va/drivers"])
@@ -100,6 +99,13 @@ PKG_CHECK_MODULES([DRM], [libdrm])
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
 
+# Check for libdrm >= 2.4 (needed for i965_drv_video.so)
+if test x$enable_i965_driver = xyes && ! $PKG_CONFIG --atleast-version=2.4 libdrm; then
+    AC_MSG_WARN([libdrm < 2.4 found, disabling build of i965 video driver])
+    enable_i965_driver=no
+fi
+AM_CONDITIONAL(BUILD_I965_DRIVER, test x$enable_i965_driver = xyes)
+
 # We only need the headers, we don't link against the DRM libraries
 LIBVA_CFLAGS="$DRM_CFLAGS"
 AC_SUBST(LIBVA_CFLAGS)
